src/cmd/build.js

```diff
@@ -12,9 +12,47 @@ import {createLogger} from '../util/logger';
 
 const log = createLogger(__filename);
 
+// Flow Types
+
+import type {OnSourceChangeFn} from '../watcher';
+import type {ExtensionManifest} from '../util/manifest';
+
+export type PackageCreatorFn =
+    (params: PackageCreatorParams) => Promise<ExtensionBuildResult>;
+
+export type ExtensionBuildResult = {
+  extensionPath: string,
+};
+
+export type PackageCreatorParams = {
+  manifestData?: ExtensionManifest,
+  sourceDir: string,
+  fileFilter: FileFilter,
+  artifactsDir: string,
+};
+
+export type BuildCmdParams = {
+  sourceDir: string,
+  artifactsDir: string,
+  asNeeded?: boolean,
+};
+
+export type BuildCmdExtraParams = {
+  manifestData?: ExtensionManifest,
+  fileFilter?: FileFilter,
```

Comment:
uhm... that's a good point

I tried that and, if I recall correctly, if you explicitly pass null (or another incompatible type) then the issue will be caught by flow as expected. 

Nevertheless it worth a more deep check, there are some scenario that I'd like to check to be sure that they are handled correctly.

I'll take a look at this (and the other similar issue in the other review comments).